packages/jest-runtime/src/index.ts

```diff
@@ -887,6 +888,42 @@ class Runtime {
       return this._environment.global.process;
     }
 
+    if (moduleName === 'module') {
+      const createRequire = (modulePath: string | URL) => {
+        const filename =
+          typeof modulePath === 'string'
+            ? modulePath
+            : fileURLToPath(modulePath);
+
+        return this._createRequireImplementation({
+          children: [],
+          exports: {},
+          filename,
+          id: filename,
+          loaded: false,
+        });
+      };
+
+      const overriddenModule = {...nativeModule};
```

Comment:
This might mess with equality checks (`require('module') === require ('module')`), but I think that's fine? If it's an easy fix go for it, if not just leaving a todo comment should be fine